### PR TITLE
[FLOSS T1] Unify spelling of man page

### PR DESCRIPTION
**Pull request by stefnotch**
Includes changes to the sed script and to the page URLs
Co-authored-by: Jan de Boer <Janldeboer@users.noreply.github.com>

Fix: https://github.com/ElektraInitiative/libelektra/issues/4567

## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [x] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

**Comment by Janldeboer** [comment]
We found the following ocurencies of `generate_manpages`, that we currently leave as is.

doc/help/CMakeLists.txt:
  28: 	generate_manpage (${NAME} COMPONENT elektra-bin)
  37: 	generate_manpage (${NAME} SECTION 7 COMPONENT elektra-doc)
src/libs/CMakeLists.txt:
  43: generate_manpage (

src/plugins/CMakeLists.txt:
  21: generate_manpage (

src/tools/elektrad/CMakeLists.txt:
  48: 	# generate manpages
  49: 	generate_manpage (kdb-run-elektrad FILENAME "${CMAKE_CURRENT_BINARY_DIR}/README.md" GENERATED_FROM


src/tools/hub-zeromq/CMakeLists.txt:
  34: 	generate_manpage (kdb-run-${tool} FILENAME ${CMAKE_CURRENT_SOURCE_DIR}/README.md)


src/tools/pythongen/CMakeLists.txt:
  49: 	generate_manpage (kdb-gen FILENAME ${CMAKE_CURRENT_SOURCE_DIR}/README.md)

src/tools/qt-gui/CMakeLists.txt:
  111: 	generate_manpage (kdb-qt-gui FILENAME ${CMAKE_CURRENT_SOURCE_DIR}/README.md COMPONENT elektra-qt-gui)

src/tools/webd/CMakeLists.txt:
  46: 	# generate manpages
  47: 	generate_manpage (kdb-run-webd FILENAME "${CMAKE_CURRENT_BINARY_DIR}/README.md" GENERATED_FROM

src/tools/website/CMakeLists.txt:
  41: 	generate_manpage (kdb-run-${tool} FILENAME "${CMAKE_CURRENT_BINARY_DIR}/README.md" GENERATED_FROM

src/tools/pythongen/CMakeLists.txt:
  49: 	generate_manpage (kdb-gen FILENAME ${CMAKE_CURRENT_SOURCE_DIR}/README.md)

src/tools/pythongen/template/template.man:
  6: .\" Template for manpages

src/tools/qt-gui/CMakeLists.txt:
  111: 	generate_manpage (kdb-qt-gui FILENAME ${CMAKE_CURRENT_SOURCE_DIR}/README.md COMPONENT elektra-qt-gui)

src/tools/webd/CMakeLists.txt:
  46: 	# generate manpages
  47: 	generate_manpage (kdb-run-webd FILENAME "${CMAKE_CURRENT_BINARY_DIR}/README.md" GENERATED_FROM

src/tools/website/CMakeLists.txt:
  41: 	generate_manpage (kdb-run-${tool} FILENAME "${CMAKE_CURRENT_BINARY_DIR}/README.md" GENERATED_FROM

**Comment by Janldeboer** [comment]
For the following occurencies, we couldn't figure out why the script does not change them:

src/include/kdbendian.h:
  54:  *   based on the austin bug tracker interface, manpages, and headers present in

src/tools/fuse/src/elektra_fuse/elektra_util.py:
  41:         #- proc:/ keys are only available through a cascading lookup (See manpage elektra-namespaces: "Keys in the namespace proc ... are ignored by kdbGet ... ")

src/tools/pythongen/template/template.man:
  6: .\" Template for manpages

**Comment by Janldeboer** [comment]
And for theses occurencies, we are unsure about how to address them:

scripts/cmake/Modules/LibAddMacros.cmake:
  614: function (generate_manpage NAME)
  660: 					MDFILE=${MDFILE} -D SOURCE_FILE=${SOURCE_FILE} -D MANPAGE=${OUTFILE} -P
  661: 					${CMAKE_SOURCE_DIR}/scripts/cmake/ElektraManPage.cmake

src/tools/website/public/assets/translations/en.json:
  10:   "APP.BREADCRUMBS.MAIN.DYN.MANPAGES": "Manpages",

**Comment by 0x6178656c** [comment]
> scripts/cmake/Modules/LibAddMacros.cmake: 614: function (generate_manpage NAME) 660: MDFILE=${MDFILE} -D SOURCE_FILE=${SOURCE_FILE} -D MANPAGE=${OUTFILE} -P 661: ${CMAKE_SOURCE_DIR}/scripts/cmake/ElektraManPage.cmake

I think it is harmless and makes sense to also change this. (There are other `generate_` functions which use snake case for split words.)

The following should work:

```sh
git ls-tree -r -z --name-only HEAD | xargs -0 sed -i 's/\bgenerate_manpage\b/generate_man_page/g'
```

---

> 660: MDFILE=${MDFILE} -D SOURCE_FILE=${SOURCE_FILE} -D MANPAGE=${OUTFILE} -P
> 661: ${CMAKE_SOURCE_DIR}/scripts/cmake/ElektraManPage.cmake

It think it is also safe and reasonable to replace `MANPAGE` by `MAN_PAGE` in

https://github.com/ElektraInitiative/libelektra/blob/e9f2cab6f743f63746bfb800a3b4d14a29fe7df6/scripts/cmake/ElektraManPage.cmake#L16

https://github.com/ElektraInitiative/libelektra/blob/e9f2cab6f743f63746bfb800a3b4d14a29fe7df6/scripts/cmake/Modules/LibAddMacros.cmake#L660

---

And of course `ManPage` in camel case is already following the new convention.

**Comment by stefnotch** [comment]
Lovely, thank you!

I applied those changes and unless I missed anything, there shouldn't be any misspellings of man pages left. This makes this PR ready to merge as soon as the pipeline passes.

File: README.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ In the GitHub repository the full documentation is available, including:
 
 You can read the documentation for the kdb tool, either
 
-- [on the Website](https://www.libelektra.org/manpages/kdb)
+- [on the Website](https://www.libelektra.org/man pages/kdb)
```
Comment (0x6178656c) [inline README.md:107]: I think the path is specified in `src/tools/website/resources/structure.json.in`

```suggestion
- [on the Website](https://www.libelektra.org/man-pages/kdb)
```

Comment (stefnotch) [reply README.md]: Nice catch, thank you very much

**Comment by flo91** [comment]
jenkins build libelektra please

**Comment by flo91** [comment]
Amazing work! 👍🏻 
What about the links to the old URLs with `manpages` instead of `man-pages`?
@stefnotch wrote about this issue at https://github.com/ElektraInitiative/libelektra/issues/4567#issuecomment-1282114421

If I didn't miss anything, the old URLs seem to be no longer valid after merging your changes.
@markus2330 How should we proceed? Looks like a redirection for the old URLs is required.

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
> If I didn't miss anything, the old URLs seem to be no longer valid after merging your changes.
> @markus2330 How should we proceed? Looks like a redirection for the old URLs is required.

As we are pre-1.0 and further big changes are planned (#4501), no redirections of old links are needed.

File: doc/news/_preparation_next_release.md [hunk 2]
```diff
@@ -282,7 +282,7 @@ This section keeps you up-to-date with the multi-language support provided by El
 - <<TODO>>
 - Fix grammar for `elektra-granularity.md` _(@dtdirect)_
 - Rephrase sections in doc/dev/error-\* _(@dtdirect)_
-- <<TODO>>
+- Unify spelling of man pages _(@stefnotch)_ _(@janldeboer)_
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:285]: Please also write in the Website section that the URL changed.

**Review by markus2330 — CHANGES_REQUESTED** [review]
Very good job, only release notes could be a bit improved.

Lets see if the CI runs through (failed due to out of disc, see #160).

**Comment by stefnotch** [comment]
I'm not entirely sure why one of the macOS builds seems to fail. 

About the merge conflicts, I can fix them and rebase this PR if desired.

**Comment by flo91** [comment]
Thank you for reporting!
I retriggered the macOS build and it succeeded. :)
Seems like it was a temporary issue.

As you already mentioned, please rebase and resolve the merge conflicts.

**Comment by flo91** [comment]
Great work, thank you! :trophy:

**Pull request merged**